collections/_posts/2019-08-22-self-hosting-a-shadowsocks-vpn-with-outline.md

```diff
@@ -0,0 +1,84 @@
+
+### Step 1 — Download & Install Outline Manager
+
+Outline allows you to setup and configure your servers from an easy-to-use management console called Outline Manager, which can be downloaded from [getoutline.org](https://getoutline.org/en/home). It has binaries available for Windows, macOS, and Linux.
+
+Simply download and install the Outline Manager application to your computer.
+
+![](/assets/img/2019-08-22-outline-install.png){: .w-100}
+
+Note: getoutline.org is blocked in China and likely other countries, however you can download the releases directly from [their GitHub page](https://github.com/Jigsaw-Code/outline-server/releases) as well.
+
+### Step 2 — Choose a Server Provider
+
+Outline has the ability to create servers on three different providers automatically: DigitalOcean, Google Cloud, and Amazon Web Services. In some situations, Google Cloud or AWS may be preferable, because they are less likely to be blocked by hostile ISPs/governments and will therefore allow you to more likely circumvent internet censorship. However, keep in mind that the server provider you choose—like any VPN provider—will have the technical ability to read your internet traffic. This is much less likely to happen when using a cloud provider versus a commercial VPN, which is why we recommend self-hosting, but it is still possible. Choose a provider you trust.
+
+Another factor to consider is your provider's network and latency. Choosing a server closer to you (geographically speaking) will give you better latency, and choosing a server with good bandwidth (>1 Gbps) will minimize the performance hit when using the VPN. Both factors are important to keeping a good browsing experience, but keep in mind using *any* VPN will always be slower than just your plain old internet connection.
+
+Finally, if you want to go with DigitalOcean you can use my affiliate link to receive a $50 credit: [https://m.do.co/c/fb6730f5bb99](https://m.do.co/c/fb6730f5bb99). That's 10 months of free VPN hosting, at $5/month/server. Don't feel obligated to use this link, but you'll receive free credit, and if you spend $25 with DigitalOcean after using it I will get credited, which will enable me to continue writing guides like this! DigitalOcean has a great performing network in my personal experience, and in the experience of the Outline team it works well in regard to circumventing censorship: Not many IP addresses of theirs are blocked.
+
+For this guide we are not going to use an automatic provider in Outline Manager, rather we will manually configure a Linux server. We are using Debian 10. Other distros may work as well, but you may need to install Docker manually.
+
+### Step 3 — Configure Your Server
+
+First, we need to install `curl`. Connect to your server via SSH and enter the following commands:
+
+```
+sudo apt update
+sudo apt install curl
```

Comment:
I would add `sudo apt upgrade` here just in case there are security updates available. And while not directly relevant, I think there should be something on hardening SSH (like requiring key authentication or at least installing sshguard/equivalent) and maybe `unattended-upgrades`.